client/app/lib/util/fetchorigintemplate.coffee

```diff
@@ -0,0 +1,11 @@
+kd = require 'kd'
+
+module.exports = fetchOriginTemplate = (stackTemplate, callback) ->
+
+  return callback null  unless stackTemplate
+
+  originalStackTemplateId = stackTemplate.config?.clonedFrom
+  return callback null  unless originalStackTemplateId
```

Comment:
on all of the remote operations we're providing the following structure on callbacks; `(err, data)` `err` can be anything about the operation itself and the `data` is the result. if an operation not produce any data it doesn't mean that there is an error. so, in application wise;

```coffee

remote.api.JSession.activeSession (err, session) ->
  if err then handleError err
  else if not session then handleNoSession
  else handleSession session
```
would be the simpler example.